plugins/pom-analyzer/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/MavenResolvedArtifactImpl.java

```diff
@@ -0,0 +1,276 @@
+        if(artifactRepositories != null) {
+            String coordinate = artifact.toString();
+            String url = getUrl(artifactResult.getRepository());
+            File file = artifact.getFile();
+            artifactRepositories.add(new String[] {coordinate, url, file.getAbsolutePath() });
+        }
+        // </new>
+
+        // SHRINKRES-143 lets ignore invalid scope
+        ScopeType scopeType = ScopeType.RUNTIME;
+        try {
+            scopeType = ScopeType.fromScopeType(root.getDependency().getScope());
+        } catch (IllegalArgumentException e) {
+            // let scope be RUNTIME
+            log.log(Level.WARNING, "Invalid scope {0} of retrieved dependency {1} will be replaced by <scope>runtime</scope>",
+                    new Object[] { root.getDependency().getScope(), root.getDependency().getArtifact() });
+        }
+
+        final List<DependencyNode> children = root.getChildren();
+        final boolean optional = root.getDependency().isOptional();
+        return new MavenResolvedArtifactImpl(artifact, scopeType, children, optional);
+    }
+
+    // <new> (3/3)
+    private static String getUrl(ArtifactRepository repository) {
+        if(repository instanceof RemoteRepository) {
+            return ((RemoteRepository)repository).getUrl();
+        } else {
+            return String.format("n/a (%s)", repository.getClass().getSimpleName());
+        }
+    }
+    // </new>
+
+    @Override
+    public <RETURNTYPE> RETURNTYPE as(Class<RETURNTYPE> returnType) {
+        if (returnType == null) {
+            throw new IllegalArgumentException("Type must be specified.");
+        }
+
+        final FormatProcessor<? super MavenResolvedArtifact, RETURNTYPE> processor = FormatProcessors.find(
+                MavenResolvedArtifact.class, returnType);
+
+        return processor.process(this, returnType);
+    }
+
+    @Override
+    public File asFile() {
+        return file;
+    }
+
+    @Override
+    public InputStream asInputStream() {
+        return as(InputStream.class);
+    }
+
+    @Override
+    public MavenResolvedArtifact asResolvedArtifact() {
+        return as(MavenResolvedArtifact.class);
+    }
+
+    @Override
+    public String toString() {
+        return "MavenResolvedArtifactImpl [mavenCoordinate=" + mavenCoordinate + ", resolvedVersion=" + resolvedVersion
+                + ", snapshotVersion=" + snapshotVersion + ", extension=" + extension + ", dependencies="
+                + Arrays.toString(dependencies) + "]";
+    }
+
+    /**
+     * Maps an artifact to a file. This allows ShrinkWrap Maven resolver to package reactor related dependencies.
+     */
+    private static File artifactToFile(final Artifact artifact) throws IllegalArgumentException {
+        if (artifact == null) {
+            throw new IllegalArgumentException("ArtifactResult must not be null");
+        }
+
+        // FIXME: this is not a safe assumption, file can have a different name
```

Comment:
Should this comment be addressed in another PR or in this one?

Reply:
This file is literally a copy of a file from Shrinkwrap. I just added several smaller patches to get access to the resolved repository. I would prefer not to fix this in our own code base. If you think this is worthwhile looking into, then you should submit a patch to Shrinkwrap :P